### PR TITLE
testcase/audio/utc: Adding tc's for new APIs

### PR DESCRIPTION
**Pull request by shivgarg**
TC for new apis in tinyalsa : pcm_drain and pcm_drop

### CODE REVIEW

**Comment by shivgarg** [comment]
@Taejun-Kwon @jsdosa @jarvisjang @kishore-sn @btheosam . This is the new PR for utc.

File: apps/examples/testcase/ta_tc/audio/utc/utc_audio_main.c [hunk 1]
```diff
@@ -763,6 +875,62 @@ static void utc_audio_pcm_writei_n(void)
...
+	/* Executed after writei positive tc */
+
+	int ret;
+
+	printf("Draining buffers to complete playback and close device\n");
+	ret = pcm_drain(g_pcm);
+	TC_ASSERT_GEQ_CLEANUP("pcm_drain", ret, 0, clean_all_data(0, NULL));
+	printf("Playback done!\n");
+
+	clean_all_data(0, NULL);
```
Comment (sunghan-chang) [inline apps/examples/testcase/ta_tc/audio/utc/utc_audio_main.c]: Why do you call it for 0, NULL?

**Comment by seinfra** [comment]
Code Rule Check OK.

File: apps/examples/testcase/ta_tc/audio/utc/utc_audio_main.c (hunk 1 above)
Comment (shivgarg) [reply apps/examples/testcase/ta_tc/audio/utc/utc_audio_main.c]: We need to close the pcm device.  In this tc no file or buffer is allocated, so clean_all_data is called with 0 and NULL.

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by manoharahk** [comment]
Dear Shivgarg,

commit title of your second patch("changing pcm_stop function to wait for deque message of all buffers t…") is too lengthy. 
commit title must be a short summary of your change.
It is better to add the description in the body of the commit message.

Also, . Signed-off-by is a must . please add "Signed-off-by" in all the commits

**Comment by seinfra** [comment]
Code Rule Check OK.

**Pull request merged**